apps/catalog-admin/pages/catalogs/[catalogId]/general/design/index.tsx

```diff
@@ -2,40 +2,69 @@ import React, { useEffect, useState } from 'react';
 import { useAdminState } from '../../../../../context/admin';
 import { ColorPicker } from '../../../../../components/color-picker';
 import { ImageUploader } from '../../../../../components/image-uploader';
-import { Banner } from '../../../../../components/banner';
 import styles from './design.module.css';
 import { Breadcrumbs, Button, PageBanner } from '@catalog-frontend/ui';
 import { Heading, HelpText, TextField } from '@digdir/design-system-react';
 import { hasOrganizationReadPermission, localization } from '@catalog-frontend/utils';
 import { getToken } from 'next-auth/jwt';
-import { Organization } from '@catalog-frontend/types';
+import { Design, Organization } from '@catalog-frontend/types';
 import { getOrganization } from '@catalog-frontend/data-access';
 import { useRouter } from 'next/router';
 import { textRegexWithNumbers } from '@catalog-frontend/utils';
+import { useGetDesign, useGetLogo, useUpdateDesign } from '../../../../../hooks/design';
+import { compare } from 'fast-json-patch';
 
 const DesignPage = ({ organization, FDK_REGISTRATION_BASE_URI }) => {
   const adminContext = useAdminState();
   const router = useRouter();
-  const catalogId: string = `${router.query.catalogId}` ?? '';
-  const pageSubtitle = organization.organization?.name ?? catalogId;
+  const catalogId = `${router.query.catalogId}` || '';
+  const pageSubtitle = organization.organization?.name || catalogId;
+
   const { backgroundColor, fontColor, logo } = adminContext;
-  const [imageLabel, setImageLabel] = useState('');
+
   const [isTextInputValid, setIsTextInputValid] = useState(false);
   const [disableTextField, setDisableTextField] = useState(true);
 
-  function validateLogoDescription(text: string): boolean {
-    return logo ? textRegexWithNumbers.test(text) : false;
-  }
+  const { data: getDesign } = useGetDesign(catalogId);
+  const dbDesign: Design = getDesign;
 
-  useEffect(() => {
-    if (logo) {
-      setDisableTextField(false);
-      setIsTextInputValid(false);
+  const { data: getLogo } = useGetLogo(catalogId);
+  const dbLogo = getLogo;
+
+  const [imageLabel, setImageLabel] = useState('');
+  const updateDesign = useUpdateDesign(catalogId);
+
+  const handleUpdateDbDesign = () => {
+    const newDesign = {
+      backgroundColor: backgroundColor,
+      fontColor: fontColor,
+      logoDescription: imageLabel,
+      hasLogo: dbDesign.hasLogo,
+    };
+
+    const diff = compare(dbDesign, newDesign);
+
+    if (diff) {
+      updateDesign
+        .mutateAsync({ oldDesign: dbDesign, newDesign: newDesign })
+        .then(() => alert(localization.alert.success))
```

Comment:
Er det tenkt at disse alertene skal senere erstattes med komponenter fra DS, eller er vanlig nodejs alert bra nok for dette formålet?

Reply:
Tror Mateus jobber med å designe en custom altert komponent!